core/retries-api/src/main/java/software/amazon/awssdk/retries/api/RetryStrategy.java

```diff
@@ -41,7 +41,10 @@
  */
 @ThreadSafe
 @SdkPublicApi
-public interface RetryStrategy extends ToCopyableBuilder<RetryStrategy.Builder, RetryStrategy> {
+public interface RetryStrategy<
+    B extends CopyableBuilder<B, T> & RetryStrategy.Builder<B, T>,
+    T extends ToCopyableBuilder<B, T> & RetryStrategy<B, T>>
+    extends ToCopyableBuilder<B, T> {
```

Comment:
We need this to be able to have to different classes extending from the builder and keep the ability to chain the builder calls.